openquake/hazardlib/gsim/nshmp_2014.py

```diff
@@ -59,6 +61,59 @@ def nga_west2_epistemic_adjustment(magnitude, distance):
         adjustment[distance < 10.0] = 0.25
     return adjustment
 
+DEFAULT_WEIGHTING = [(0.185, -1.), (0.63, 0.), (0.185, 1.)]
+
+
+def get_weighted_poes(gsim, sctx, rctx, dctx, imt, imls, truncation_level,
+                      weighting=DEFAULT_WEIGHTING):
+    """
+    This function implements the NGA West 2 GMPE epistemic uncertainty
+    adjustment factor without re-calculating the actual GMPE each time.
+    :param gsim:
+        Instance of the GMPE
+    :param list weighting:
+        Weightings as a list of tuples of (weight, number standard deviations
+        of the epistemic uncertainty adjustment)
+    """
+    if truncation_level is not None and truncation_level < 0:
+        raise ValueError('truncation level must be zero, positive number '
+                         'or None')
+    gsim._check_imt(imt)
+    # Boore et al. (2014) has only rjb - catch this case
+    if hasattr(dctx, "rrup"):
+        adjustment = nga_west2_epistemic_adjustment(rctx.mag, dctx.rrup)
```

Comment:
Yes, my approach is not the best solution here. I will modify it so that it calculates rrup for the Boore et al GMPE too and use the rrup values rather than rjb